src/requestBuilder.ts

```diff
@@ -61,14 +61,30 @@ function getToolRequest(
   if (!Number.isInteger(sessionCounter)) {
     sessionCounter = 0
   }
+  if (client) {
+    // Determine if the session is engaged to set the 'seg' value
+    const pageviewCounter = parseInt(client.get('pageviewCounter') || '0')
+    const conversionCounter = parseInt(client.get('conversionCounter') || '0')
+    const engagementDuration = parseInt(client.get('engagementDuration') || '0')
+
+    // Session will be marked engaged if longer than 10 seconds, has at least 1 conversion event, and 2 or more pageviews
+    if (
+      engagementDuration > 10 &&
```

Comment:
```suggestion
      engagementDuration >= 10 &&
```
Example in docs here https://support.google.com/analytics/answer/12195621?hl=en&ref_topic=11151952&sjid=11966088143636941671-EU suggests that it's 10 or more seconds